### ISSUE TITLE
SQS receive message MessageAttributes part of Body

### ISSUE THREAD

**Issue by puppeteer701** [localstack/localstack#3725]
[x] bug report

Localstack: latest version - 0.12.7

On sqs receive message, `MessageAttributes` is part of a body.

```
aws sns publish --endpoint-url=http://127.0.0.1:4566 --topic-arn "arn:aws:sns:eu-west-2:000000000000:bananas" --message '{"headers":{"type":"success","version":1},"payload":{"modulr":{"id":"11111"},"internal":{"id":1,"type":"1dd"}}}' --message-attributes 'route={DataType=String,StringValue="something"}'
```

Running a golang server with `MessageAttributeNames=[]*string{All}`

```
 aws sqs receive-message --endpoint-url=http://127.0.0.1:4566 --queue-url http://127.0.0.1:4566/queue/bananas

{
    "Messages": [
        {
            "MessageId": "c5db0b55-cee9-a8a6-c022-4aa7f634de86",
            "ReceiptHandle": "sgnunynfxpwrrqiiiywpwnllzeedkksydbhfgjoiyqgvknxtpofomyexbxmnwhqwqoijpugdwgwijkuvvwiwxwslcxckujzoixohaakhnlaawtaodveejojestrgnqtkumzajjaitoivakjjrwggehljlyoemuqzljhgnsadgftgpfvipmwinyhkb",
            "MD5OfBody": "cfd9aac444d6cfce27a2cc2386a172e4",
            "Body": "{\"Type\": \"Notification\", \"MessageId\": \"ae4f517d-c1aa-4559-9e70-9308b71515fb\", \"TopicArn\": \"arn:aws:sns:eu-west-2:000000000000:bananas\", \"Message\": \"{\\\"headers\\\":{\\\"type\\\":\\\"bananas\\\",\\\"version\\\":1},\\\"payload\\\":{\\\"modulr\\\":{\\\"PaymentId\\\":\\\"P434343\\\"},\\\"internal\\\":{\\\"id\\\":1,\\\"type\\\":\\\"WDID\\\"}}}\", \"Timestamp\": \"2021-03-14T22:32:12.643Z\", \"SignatureVersion\": \"1\", \"Signature\": \"EXAMPLEpH+..\", \"SigningCertURL\": \"https://sns.us-east-1.amazonaws.com/SimpleNotificationService-0000000000000000000000.pem\", \"MessageAttributes\": {\"route\": {\"Type\": \"String\", \"Value\": \"something\"}}}"
        }
    ]
}
```

Docs of aws suggest it should be part of `Message` not `Message.Body`.

https://docs.aws.amazon.com/cli/latest/reference/sqs/receive-message.html

```
  "Messages": [
    {
      "Body": "My first message.",
      "ReceiptHandle": "AQEBzbVv...fqNzFw==",
      "MD5OfBody": "1000f835...a35411fa",
      "MD5OfMessageAttributes": "9424c491...26bc3ae7",
      "MessageId": "d6790f8d-d575-4f01-bc51-40122EXAMPLE",
      "Attributes": {
        "ApproximateFirstReceiveTimestamp": "1442428276921",
        "SenderId": "AIDAIAZKMSNQ7TEXAMPLE",
        "ApproximateReceiveCount": "5",
        "SentTimestamp": "1442428276921"
      },
      "MessageAttributes": {
        "PostalCode": {
          "DataType": "String",
          "StringValue": "ABC123"
        },
        "City": {
          "DataType": "String",
          "StringValue": "Any City"
        }
      }
    }
  ]
}
```

**Comment by HammurabisCoder** [comment]
I've run into this too using the AWS SDK to connect to Localstack. Something else to note in addition to being in the body is that with AWS SQS we can read the attributes with `.StringValue` while LocalStack is just `.Value`.

**Comment by whummer** [comment]
Thanks for reporting @puppeteer701 @HammurabisCoder . This should be fixed in #3887 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!
